### ISSUE TITLE
Confusing error messages while attempting to run aggregations on ip fields

### ISSUE THREAD

**Issue by imotov** [quickwit-inc/tantivy#1835]
**Describe the bug**
Ip fields are not supported by aggregations at the moment, but trying to run aggregations on ip field are causing confusing error messages such as  "Unknown code`{code}.`'" error or "panicked at 'attempt to shift left with overflow'"

**Which version of tantivy are you using?**
0f2078791783321020b441f7333971939bdd9a04

**To Reproduce**

Here is a test that demonstrates the problem https://github.com/imotov/tantivy/commit/035db34734b3eb91438dea8590c42844fbbf15be

You can also reproduce this issues by creating an ip field in quickwit and trying to run aggregations on it. It produces a slightly different but still confusing error message.

**Comment by fulmicoton** [comment]
Any panic is more than an error message, it's a hard bug.

@PSeitz can you have a look?

**Comment by imotov** [comment]
I just want to add that I can no longer reproduce the "Unknown code{code}.'" error with quickwit on today's main branch. I am getting panic now with 'attempt to shift left with overflow', which is the same error I get in the tent ivy-only test I've attached.

**Comment by PSeitz** [comment]
When using the main branch, you probably need to reindex, since the format compatibility will break for the next release

**Comment by imotov** [comment]
I think I figured it out. I tested using this script https://github.com/imotov/qwscripts . If you run this script in the source directory it builds everything from source, but you run this script in any other directory it uses a docker image. The `test-load.zsh` deletes the index, recreates it and starts a searcher on it. The `test-search.zsh` runs queries. 

When I run in on the docker image I get the "Unknown code{code}." error, when I run it on source code I get "panicked at 'attempt to shift left with overflow'". The docker image is `Quickwit 0.4.0-nightly (cd64d15 2023-01-20T16:50:40Z)`.
